packages/flame/lib/src/components/sprite_animation_component.dart

```diff
@@ -35,11 +35,12 @@ class SpriteAnimationComponent extends PositionComponent {
     animation = SpriteAnimation.fromFrameData(image, data);
   }
 
-  /// Component will be removed after loop end and [removeOnFinish] is set.
+  /// Component will be removed after animation is done and [removeOnFinish] is set.
+  ///
+  /// Note: [SpriteAnimationComponent] will not be removed automatically if loop property of [SpriteAnimation] is true.
   @override
   bool get shouldRemove =>
-      super.shouldRemove ||
-      (removeOnFinish && (animation?.isLastFrame ?? false));
+      super.shouldRemove || (removeOnFinish && (animation?.done() ?? false));
```

Comment:
Would be nice to have an unit test here, to ensure this behaviour in the future

Reply:
> Would be nice to have an unit test here, to ensure this behaviour in the future

I will try to add a unit test later.